### ISSUE TITLE
Essential task `grandpa-voter` failed. Shutting down service. Error: Input("Essential task failed.")

### ISSUE THREAD

**Issue by HackFisher** [darwinia-network/darwinia#560]
[long_text_2020-10-09-10-40-56.txt](https://github.com/darwinia-network/darwinia/files/5353338/long_text_2020-10-09-10-40-56.txt)

**Comment by wi1dcard** [comment]
Hint: Only the validator nodes that **prepare** new blocks got this issue. Other nodes like the full node, or the validator node that doesn't prepare new blocks don't.

**Comment by HackFisher** [comment]
Related: https://github.com/paritytech/substrate/issues/6727

**Comment by wi1dcard** [comment]
## Workarounds for `Too many open files`

Run the node with Docker:

```bash
docker run --ulimit nofile=102400:102400 darwinianetwork/darwinia ...
```

Run the node with Systemd, edit your service unit file, add:

```
[Service]
LimitNOFILE=102400
```

Run the node with shell, PM2, or any PAM authenticated user, edit `/etc/security/limits.conf`, add:

```
root soft nofile 65535
root hard nofile 65535
* soft nofile 65535
* hard nofile 65535
```

**Comment by AurevoirXavier** [comment]
https://github.com/darwinia-network/darwinia-common/pull/323

**Comment by wuminzhe** [comment]
docker does not have this problem, the docker image seems to provide a large number: 1048576

**Comment by AurevoirXavier** [comment]
Will be fixed in 0.7.2.

Reopen if needed.
